chain/chain/src/chain.rs

```diff
@@ -3067,15 +3087,29 @@ impl<'a> ChainUpdate<'a> {
         }
 
         // Verify that proposals from chunks match block header proposals.
-        let mut all_chunk_proposals = vec![];
-        for chunk in block.chunks().iter() {
-            if block.header().height() == chunk.height_included() {
-                all_chunk_proposals.extend(chunk.validator_proposals().iter().cloned());
+        let block_height = block.header().height();
+        for (hp, cp) in utils::full_zip(
+            block.header().validator_proposals(),
+            block
+                .chunks()
+                .iter()
+                .filter(|chunk| block_height == chunk.height_included())
+                .flat_map(|chunk| chunk.validator_proposals()),
+        ) {
+            match (hp, cp) {
+                (None, Some(_)) | (Some(_), None) | (None, None) => {
+                    // Can only occur if there were a different number of proposals in the header
+                    // and chunks
+                    return Err(ErrorKind::InvalidValidatorProposals.into());
+                }
+                (Some(hp), Some(cp)) => {
```

Comment:
If you switch the order of the two cases you can simply use `_` for the second case

Reply:
Done in https://github.com/near/nearcore/pull/4115/commits/70bc050f9bdc187030da0ba5795b3bf646ac46ed